x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -5,13 +5,56 @@
  * 2.0.
  */
 
+import dateMath from '@elastic/datemath';
 import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
-import { chunk } from 'lodash';
-import { ALERT_UUID, VERSION } from '@kbn/rule-data-utils';
+import { RuleExecutorOptions } from '@kbn/alerting-plugin/server';
+import { chunk, partition } from 'lodash';
+import {
+  ALERT_INSTANCE_ID,
+  ALERT_LAST_DETECTED,
+  ALERT_SUPPRESSION_DOCS_COUNT,
+  ALERT_SUPPRESSION_END,
+  ALERT_UUID,
+  ALERT_WORKFLOW_STATUS,
+  TIMESTAMP,
+  VERSION,
+} from '@kbn/rule-data-utils';
 import { getCommonAlertFields } from './get_common_alert_fields';
 import { CreatePersistenceRuleTypeWrapper } from './persistence_types';
 import { errorAggregator } from './utils';
 import { createGetSummarizedAlertsFn } from './create_get_summarized_alerts_fn';
+import { AlertWithSuppressionFields870 } from '../../common/schemas/8.7.0';
+
+const augmentAlerts = <T>({
+  alerts,
+  options,
+  kibanaVersion,
+  includeLastDetected,
+  currentTimeOverride,
+}: {
+  alerts: Array<{ _id: string; _source: T }>;
+  options: RuleExecutorOptions<any, any, any, any, any>;
+  kibanaVersion: string;
+  includeLastDetected: boolean;
+  currentTimeOverride: Date | undefined;
+}) => {
+  const commonRuleFields = getCommonAlertFields(options);
+  return alerts.map((alert) => {
+    return {
+      ...alert,
+      _source: {
+        [ALERT_LAST_DETECTED]: includeLastDetected ? currentTimeOverride ?? new Date() : undefined,
+        [VERSION]: kibanaVersion,
+        ...commonRuleFields,
+        ...alert._source,
```

Comment:
We originally chose to override the common fields with the passed in alert fields because we wanted to override some fields like `TAGS`. Historically the security solution added `__internal__` tags to rules that we used for sorting/filtering rules, but we wanted to avoid exposing those to users as much as possible. It appears we removed those tags entirely in https://github.com/elastic/kibana/pull/130664. (although, looking closer, we use `kibana.alert.rule.tags` in the security solution and the common field is just `tags` so I don't think we actually overrode anything).

We also override `ALERT_RULE_PARAMETERS` with a slightly different format at the moment - we convert the params to snake case before adding them to the alert so that the case is consistently snake case throughout the alert doc. `ALERT_RULE_NAME` and `TIMESTAMP` are also overridden in some cases with different values from the security solution.

I'm definitely open to exploring ways we can improve the way we build these alert documents up. It would be convenient to define the alert type passed into `alertWithPersistence` with `never` restrictions like `T extends { [TAGS]: never }` so fields don't get overridden by surprise. I think we'd still want the ability to pass in explicit overrides for these common fields though.

Reply:
Sounds good, thank you for the context! We'll revise this outside this PR as needed 🙏